pandas/tests/tseries/test_timezones.py

```diff
@@ -1728,6 +1506,245 @@ def test_nat(self):
+        ts2 = Series(np.random.randn(len(rng2)), index=rng2)
+        ts_result = ts1.append(ts2)
+        assert ts_result.index.equals(ts1.index.asobject.append(
+            ts2.index))
+
+    def test_append_dst(self):
+        rng1 = date_range('1/1/2016 01:00', periods=3, freq='H',
+                          tz='US/Eastern')
+        rng2 = date_range('8/1/2016 01:00', periods=3, freq='H',
+                          tz='US/Eastern')
+        ts1 = Series([1, 2, 3], index=rng1)
+        ts2 = Series([10, 11, 12], index=rng2)
+        ts_result = ts1.append(ts2)
+
+        exp_index = DatetimeIndex(['2016-01-01 01:00', '2016-01-01 02:00',
+                                   '2016-01-01 03:00', '2016-08-01 01:00',
+                                   '2016-08-01 02:00', '2016-08-01 03:00'],
+                                  tz='US/Eastern')
+        exp = Series([1, 2, 3, 10, 11, 12], index=exp_index)
+        assert_series_equal(ts_result, exp)
+        assert ts_result.index.tz == rng1.tz
+
+    def test_append_aware(self):
+        rng1 = date_range('1/1/2011 01:00', periods=1, freq='H',
+                          tz='US/Eastern')
+        rng2 = date_range('1/1/2011 02:00', periods=1, freq='H',
+                          tz='US/Eastern')
+        ts1 = Series([1], index=rng1)
+        ts2 = Series([2], index=rng2)
+        ts_result = ts1.append(ts2)
+
+        exp_index = DatetimeIndex(['2011-01-01 01:00', '2011-01-01 02:00'],
+                                  tz='US/Eastern')
+        exp = Series([1, 2], index=exp_index)
+        assert_series_equal(ts_result, exp)
+        assert ts_result.index.tz == rng1.tz
+
+        rng1 = date_range('1/1/2011 01:00', periods=1, freq='H', tz='UTC')
+        rng2 = date_range('1/1/2011 02:00', periods=1, freq='H', tz='UTC')
+        ts1 = Series([1], index=rng1)
+        ts2 = Series([2], index=rng2)
+        ts_result = ts1.append(ts2)
+
+        exp_index = DatetimeIndex(['2011-01-01 01:00', '2011-01-01 02:00'],
+                                  tz='UTC')
+        exp = Series([1, 2], index=exp_index)
+        assert_series_equal(ts_result, exp)
+        utc = rng1.tz
+        assert utc == ts_result.index.tz
+
+        # GH 7795
+        # different tz coerces to object dtype, not UTC
+        rng1 = date_range('1/1/2011 01:00', periods=1, freq='H',
+                          tz='US/Eastern')
+        rng2 = date_range('1/1/2011 02:00', periods=1, freq='H',
+                          tz='US/Central')
+        ts1 = Series([1], index=rng1)
+        ts2 = Series([2], index=rng2)
+        ts_result = ts1.append(ts2)
+        exp_index = Index([Timestamp('1/1/2011 01:00', tz='US/Eastern'),
+                           Timestamp('1/1/2011 02:00', tz='US/Central')])
+        exp = Series([1, 2], index=exp_index)
+        assert_series_equal(ts_result, exp)
+
+
+class TestDataFrameTimeZones(object):
+    timezones = ['UTC', 'Asia/Tokyo', 'US/Eastern', 'dateutil/US/Pacific']
```

Comment:
what is this for?

Reply:
There are a bunch of GH issues related to timezones/timeseries/... and many of them involve behavior that is specific to DataFrame/Series/FooIndex.  The goal for this PR (and presumably its a multi-PR goal) is to separate these cases to test in isolation.

Reply:
i.e. this just collects all of the test_timezones tests that use pd.DataFrame.